src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T[]): GroupsMap<T> {
+  const keys: string| number[] = [];
+
+  items.map((item) => (keys.includes(item[key])
+    ? null
+    : keys.push(item[key])));
```

Comment:
```suggestion
  items.forEach((item) => (!keys.includes(item[key]) && keys.push(item[key])));
```
better to use forEach 'vause you don't need the array that map returns